### PR TITLE
Add {Baby,Mama,Papa}Bear reference implementations

### LINKED ISSUE #127: Add Round 2 KEM ThreeBears

### PR DESCRIPTION
**Pull request by leonbotros**
This PR adds the reference implementation of ThreeBears by Mike Hamburg.
The code that generates these implementations is adjusted from [rainbow-clean](https://github.com/mkannwischer/rainbow-clean) and available at: [threebears-clean](https://github.com/leonbotros/threebears-clean). 
The implementations match the newest version (see, https://sourceforge.net/projects/threebears/), 
that includes implicit rejection. Feedback is welcome!

Closes #127. 

#### Manually checked properties

* [X] No stringification macros
* [X] Output-parameter pointers in functions are on the left
* [ ] Negative return values on failure of API functions (within restrictions of FO transform).
* [X] `const` arguments are labeled as `const`
* [X] variable declarations at the beginning (except in `for (size_t i=...`)
* Optional:
  * [ ] All integer types are of fixed size, using `stdint.h` types (including `uint8_t` instead of `unsigned char`)
  * [X] Integers used for indexing are of size `size_t`

### CODE REVIEW

File: crypto_kem/babybear/clean/ring.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/* Multiply and accumulate c = c + a*b */
+void PQCLEAN_BABYBEAR_CLEAN_mac(gf_t c, const gf_t a, const gf_t b);
+
+/* Reduce ring element to canonical form */
+void PQCLEAN_BABYBEAR_CLEAN_canon(gf_t c);
+
+/** Return the i'th limb of the modulus */
+static inline limb_t modulus(size_t i) {
+    return (i == DIGITS / 2) ? LMASK - 1 : LMASK;
+}
```
Comment (mkannwischer) [inline crypto_kem/babybear/clean/ring.h]: This should be moved out of the header file.

File: crypto_kem/babybear/clean/Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# This Makefile can be used with GNU Make or BSD Make
+
+LIB=libbabybear_clean.a
+
+HEADERS = api.h melas_fec.h params.h ring.h threebears.h
+OBJECTS = kem.o melas_fec.o ring.o threebears.o 
+
+
+CFLAGS=-O3 -Wcast-align=strict -Wall -Wconversion -Wextra -Wpedantic -Wvla -Wmissing-prototypes -Wredundant-decls -std=c99 -I../../../common $(EXTRAFLAGS)
```
Comment (mkannwischer) [inline crypto_kem/babybear/clean/Makefile]: @thomwiggers, does this mean that -Wcast-align=strict is now supported by all of our targets?
If yes, then we should add this to all the implementations.

Comment (thomwiggers) [reply crypto_kem/babybear/clean/Makefile]: I don't think it is; there's still false positives probably.

Comment (leonbotros) [reply crypto_kem/babybear/clean/Makefile]: I think `-Werror` is missing here

**Comment by mkannwischer** [comment]
LGTM, thanks @leonbotros!

**Comment by thomwiggers** [comment]
Travis-CI's arm64 builds appear to be working again, let's hope they get this show on the road.

**Pull request merged**